src/state/routing/useRoutingAPITrade.ts

```diff
@@ -87,19 +87,24 @@ export function useRoutingAPITrade<TTradeType extends TradeType>(
   useGetQuoteQuery(skipFetch ? skipToken : queryArgs, {
     // Price-fetching is informational and costly, so it's done less frequently.
     pollingInterval: routerPreference === INTERNAL_ROUTER_PREFERENCE_PRICE ? ms(`1m`) : AVERAGE_L1_BLOCK_TIME,
-    // If latest quote from cache was fetched > 2m ago, instantly repoll for another instead of waiting for next poll period
+    // If latest quote from cache was fetched > 2m ago, instantly repoll for another instead of waiting for next poll period.
     refetchOnMountOrArgChange: 2 * 60,
   })
   const isFetching = currentData !== tradeResult || !currentData
+  // Only "progressive" trades - those with the same currencies as the last trade - should show the last trade while loading.
+  const isProgressive =
+    tradeResult?.trade &&
+    currencyIn?.equals(tradeResult.trade.inputAmount.currency) &&
+    currencyOut?.equals(tradeResult.trade.outputAmount.currency)
 
   return useMemo(() => {
     if (amountSpecified && queryArgs === skipToken) {
       return {
         state: TradeState.STALE,
-        trade: tradeResult?.trade,
+        trade: isProgressive ? tradeResult?.trade : undefined,
         swapQuoteLatency: tradeResult?.latencyMs,
       }
-    } else if (!amountSpecified || isError || queryArgs === skipToken) {
```

Comment:
just so my understanding is correct - the check for `!amountSpecified` is redundant because if that is true than `queryArgs === skipToken` correct?